### PR TITLE
As a user or owner i want to create an account

### LINKED ISSUE #2: As a User or Owner i want to create an account
- [x] Create DTO objects
- [x] Implement controller layer
- [ ] Implement service layer

### PR DESCRIPTION
**Pull request by matsior**

### CODE REVIEW

File: src/main/java/pl/reservo/reservoservice/api/CompanyRegistrationRequest.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package pl.reservo.reservoservice.api;
+
+import lombok.Builder;
+
+import javax.validation.Valid;
+import javax.validation.constraints.Email;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+
+@Builder
```
Comment (Norsu296) [inline src/main/java/pl/reservo/reservoservice/api/CompanyRegistrationRequest.java]: If it is request, did we need builder? Can it be record?

File: src/main/java/pl/reservo/reservoservice/api/CompanyRegistrationResponse.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package pl.reservo.reservoservice.api;
+
+import lombok.Builder;
+
+@Builder
```
Comment (Norsu296) [inline src/main/java/pl/reservo/reservoservice/api/CompanyRegistrationResponse.java]: Add `@Value` annotation from lombok and remove private final

Comment (matsior) [reply src/main/java/pl/reservo/reservoservice/api/CompanyRegistrationResponse.java]: I converted class into record

File: src/main/java/pl/reservo/reservoservice/api/CompanyRegistrationRequest.java (hunk 1 above)
Comment (matsior) [reply src/main/java/pl/reservo/reservoservice/api/CompanyRegistrationRequest.java]: Changed to record

**Review by Norsu296 — APPROVED** [review]

**Pull request merged**